wire-library/wire-schema/src/jvmMain/kotlin/com/squareup/wire/schema/internal/UtilJVM.kt

```diff
@@ -28,3 +38,24 @@ internal actual inline fun String.toEnglishLowerCase() = toLowerCase(Locale.US)
 actual typealias MutableQueue<T> = Queue<T>
 
 internal actual fun <T : Any> mutableQueueOf(): MutableQueue<T> = ArrayDeque()
+
+private val UNICODE_BOMS = Options.of(
+    "efbbbf".decodeHex(), // UTF-8
+    "feff".decodeHex(), // UTF-16BE
+    "fffe".decodeHex(), // UTF-16LE
+    "0000ffff".decodeHex(), // UTF-32BE
+    "ffff0000".decodeHex() // UTF-32LE
+)
+
+@Throws(IOException::class)
+fun BufferedSource.readBomAsCharset(default: Charset = UTF_8): Charset {
```

Comment:
I'm going to move this so it's not public. It was being used in two places but now it's only one so it can just live close to its site of use.

@swankjesse should this live in Okio? Maybe as `{Path|File}.bomAwareSource()` or something?

Reply:
Not yet. The problem is Okio sources don’t know how to read anything other than UTF-8. If we ever were to add such support? That’d be a lot of complexity.

I was hoping to just wait out the universe until UTF-8 was the only thing that existed.

Reply:
Nevermind that can't be the signature 🤦 

Maybe `BufferedSource.readStringBomAware()`, although that's a mouthful.

Reply:
Even if UTF-8 is everywhere you still need to conditionally handle the presence of its BOM